pallet/staking/src/benchmarking.rs

```diff
@@ -0,0 +1,209 @@
+// This file is part of Darwinia.
+//
+// Copyright (C) 2018-2023 Darwinia Network
+// SPDX-License-Identifier: GPL-3.0
+//
+// Darwinia is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Darwinia is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Darwinia. If not, see <https://www.gnu.org/licenses/>.
+
+// darwinia
+use crate::*;
+use darwinia_deposit::SimpleAsset;
+use dc_types::UNIT;
+// substrate
+use frame_benchmarking::v2::{self, Linear};
+use frame_system::RawOrigin;
+use sp_std::prelude::*;
+
+#[v2::benchmarks]
+mod benchmarks {
+	// darwinia
+	use super::*;
+
+	fn deposit_for<T>(who: &T::AccountId, count: u32) -> Vec<DepositId<T>>
+	where
+		T: Config + darwinia_deposit::Config,
+	{
+		(0..count.min(<<T as Config>::MaxDeposits>::get()) as u8)
+			.map(|x| {
+				<darwinia_deposit::Pallet<T>>::lock(
+					RawOrigin::Signed(who.to_owned()).into(),
+					UNIT,
+					1,
+				)
+				.unwrap();
+
+				x.into()
+			})
+			.collect()
+	}
+
+	#[benchmark]
+	fn stake(x: Linear<0, 255>) {
+		let a = frame_benchmarking::whitelisted_caller();
+
+		T::RingCurrency::make_free_balance_be(&a, 256 * UNIT);
+		T::KtonHelper::mint(&a, UNIT).unwrap();
```

Comment:
Suggest replacing `KtonHelper` with `type Kton = KtonMinting` of the `darwinia-deposit` config. They are the same in your code.

Reply:
Great idea.